packages/sharing-smart-contract/test/invariant/handlers/HandlerRenting.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.24;
+
+import {IERC721} from "forge-std/interfaces/IERC721.sol";
+import {Test} from "forge-std/Test.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {HandlerCollection} from "./HandlerCollection.sol";
+import {IRental} from "../../../contracts/interfaces/IRental.sol";
+
+contract HandlerRenting is Test, HandlerCollection {
```

Comment:
While I am in favor of organizing contracts with inheritance / composition, having handlers inherit `HandlerCollection` can be problematic: each handler will create their own data sets and will use it for its own purpose. For example, here, one data set can only be put for renting and rented, but this very same dataset will never be sold because the ghost tracking of `HandlerRenting is HandlerCollection` is different than `HandlerSale is HandlerCollection`.

It's like having a 12-cylinder sportscar that can simulate a gazillion scenarios, and you are driving it first gear :-)